kong/plugins/key-auth/handler.lua

```diff
@@ -155,6 +156,15 @@ end
 function KeyAuthHandler:access(conf)
   KeyAuthHandler.super.access(self)
 
+  -- check if preflight request and whether it should be authenticated
+  if conf.run_on_preflight == false and get_method() == "OPTIONS" then
+    -- FIXME: the above `== false` test is because existing entries in the db will
+    -- have `nil` and hence will by default start passing the preflight request
+    -- This should be fixed by a migration to update the actual entries
+    -- in the datastore
```

Comment:
I thought our agreed upon policy was to write migrations for such newly introduced fields instead of slowly digging ourselves into a hole of backwards-compatibility edge-cases?

Reply:
the PR's are separated. This goes without migration, and hence can land in `0.11.x`, whereas the accompanying PR with the migration, and removal of this TODO can land in `0.12`. (previously discussed)

Reply:
There is nowhere in this PR a reference to that other PR you are mentioning. Could you provide a link to it?

Reply:
added to the initial message

Reply:
> removal of this TODO can land in 0.12.

It would be nice if we decided on a convention on how to tag such temporary measures in the codebase, so that we grep through the code before a new version and make sure that all of these were taken care of (something more specific than `FIXME`).